ab2d-fhir/src/main/java/gov/cms/ab2d/fhir/IdentifierUtils.java

```diff
@@ -206,41 +207,58 @@ private static PatientIdentifier.Currency getCurrencyMbiStandard(ICompositeType
             return PatientIdentifier.Currency.UNKNOWN;
         }
+
         Object val = vals.get(0);
         String codeSystem = (String) Versions.invokeGetMethod(val, GET_SYSTEM);
         String codeValue = (String) Versions.invokeGetMethod(val, GET_CODE);
-        if (codeSystem != null && codeSystem.equalsIgnoreCase(MBI_ID_R4) && ("MB".equalsIgnoreCase(codeValue) || "MC".equalsIgnoreCase(codeValue))) {
-            List extensions = (List) Versions.invokeGetMethod(val, "getExtension");
-            if (extensions != null && extensions.size() > 0) {
-                Object extension = extensions.get(0);
-                String url = (String) Versions.invokeGetMethod(extension, "getUrl");
-                if (url != null && url.equalsIgnoreCase(CURRENCY_IDENTIFIER)) {
-                    Object currValue = Versions.invokeGetMethod(extension, GET_VALUE);
-                    String extValueSystem = (String) Versions.invokeGetMethod(currValue, GET_SYSTEM);
-                    if (CURRENCY_IDENTIFIER.equalsIgnoreCase(extValueSystem)) {
-                        String currValueCode = (String) Versions.invokeGetMethod(currValue, GET_CODE);
-                        if (CURRENT_MBI.equalsIgnoreCase(currValueCode)) {
-                            return PatientIdentifier.Currency.CURRENT;
-                        }
-                        if (HISTORIC_MBI.equalsIgnoreCase(currValueCode)) {
-                            return PatientIdentifier.Currency.HISTORIC;
-                        }
-                    }
-                }
+        if (!checkCodingIsValid(codeSystem, codeValue)) {
+            return PatientIdentifier.Currency.UNKNOWN;
+        }
+
+        List extensions = (List) Versions.invokeGetMethod(val, "getExtension");
+        if (extensions == null || extensions.isEmpty()) {
+            return PatientIdentifier.Currency.UNKNOWN;
+        }
+
+        Object extension = extensions.get(0);
+        String url = (String) Versions.invokeGetMethod(extension, "getUrl");
+        if (!checkCurrencyUrlIsValid(url)) {
+            return PatientIdentifier.Currency.UNKNOWN;
+        }
+        
+        Object currValue = Versions.invokeGetMethod(extension, GET_VALUE);
+        String extValueSystem = (String) Versions.invokeGetMethod(currValue, GET_SYSTEM);
+        if (CURRENCY_IDENTIFIER.equalsIgnoreCase(extValueSystem)) {
+            String currValueCode = (String) Versions.invokeGetMethod(currValue, GET_CODE);
+            if (CURRENT_MBI.equalsIgnoreCase(currValueCode)) {
+                return PatientIdentifier.Currency.CURRENT;
+            }
+            if (HISTORIC_MBI.equalsIgnoreCase(currValueCode)) {
+                return PatientIdentifier.Currency.HISTORIC;
             }
         }
         return PatientIdentifier.Currency.UNKNOWN;
     }
 
+    public static boolean checkTypeAndCodingExists(Object type, List<Coding> vals) {
+        return (type != null && vals != null && !vals.isEmpty());
```

Comment:
It's not easy to read code when you check positive case (Exists) using negative conditions and then you use `not` operator to check that case is negative
I mean something like that instead 
`public static boolean checkTypeAndCodingDontExists(Object type, List<Coding> vals) {
        return (type == null || vals == null || vals.isEmpty());
}`

Reply:
It's a bit tricky because sonarqube complains about a codesmell if I do not have these as a one-liner with a return statement followed by the conditional. So, it is hard to make every condition use the "positive" case (without using '!' anywhere). 

I am about to push a change that at least makes all of the functions checking for invalid things so that in the if statements in the function I am not doing '!' in front of all of them.

If there is anything else I can do here, maybe I am missing something obvious please let me know. Thank you!